### ISSUE TITLE
OL8: Leapp preupgrade fails on check_oracle_enabled_repos

### ISSUE THREAD

**Issue by ku4eto** [oracle/oracle-linux#142]
Hello,

not sure if this issue should be raised here or at the `leapp-repository` git.

Running `leapp preupgrade` results in the following error:

```
2024-05-19 12:57:57.69  INFO     PID: 12672 leapp.workflow.Checks: Executing actor oracle_signed_rpm_check 
2024-05-19 12:57:57.166 WARNING  PID: 17477 leapp.reporting: Stable Key report entry not provided, dynamically generating one - f5a5d58476a97bf0a8904d00df5d1321189849ad
2024-05-19 12:57:57.179 INFO     PID: 12672 leapp.workflow.Checks: Executing actor check_rdma 
2024-05-19 12:57:57.313 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_baseos_latest
2024-05-19 12:57:57.317 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_appstream
2024-05-19 12:57:57.318 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_codeready_builder
2024-05-19 12:57:57.319 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_distro_builder
2024-05-19 12:57:57.320 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_u0_baseos_base
2024-05-19 12:57:57.321 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_u1_baseos_base
2024-05-19 12:57:57.322 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_u2_baseos_base
2024-05-19 12:57:57.323 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_u3_baseos_base
2024-05-19 12:57:57.324 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_u4_baseos_base
2024-05-19 12:57:57.325 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_u5_baseos_base
2024-05-19 12:57:57.326 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_u6_baseos_base
2024-05-19 12:57:57.327 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_u7_baseos_base
2024-05-19 12:57:57.328 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_u8_baseos_base
2024-05-19 12:57:57.329 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_u9_baseos_base
2024-05-19 12:57:57.329 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_addons
2024-05-19 12:57:57.330 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_u4_security_validation
2024-05-19 12:57:57.331 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_MODRHCK
2024-05-19 12:57:57.332 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_UEKR7
2024-05-19 12:57:57.333 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_UEKR6
2024-05-19 12:57:57.335 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_UEKR7_RDMA
2024-05-19 12:57:57.336 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_UEKR6_RDMA
2024-05-19 12:57:57.337 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_kvm_appstream
2024-05-19 12:57:57.338 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_developer_EPEL
2024-05-19 12:57:57.339 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol8_developer_EPEL_modular
2024-05-19 12:57:57.340 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol9_baseos_latest
2024-05-19 12:57:57.341 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol9_appstream
2024-05-19 12:57:57.342 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol9_addons
2024-05-19 12:57:57.343 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol9_codeready_builder
2024-05-19 12:57:57.344 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol9_UEKR7
2024-05-19 12:57:57.345 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol9_RDMA
2024-05-19 12:57:57.346 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol9_kvm_utils
2024-05-19 12:57:57.347 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol9_oci_included
2024-05-19 12:57:57.348 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol9_ksplice
2024-05-19 12:57:57.349 WARNING  PID: 17526 leapp.workflow.Checks.check_rdma: REPO ol9_oraclelinuxmanager210_client
2024-05-19 12:57:57.350 INFO     PID: 17526 leapp.workflow.Checks.check_rdma: No Oracle RDMA configuration found
2024-05-19 12:57:57.360 INFO     PID: 12672 leapp.workflow.Checks: Executing actor check_mount_options 
2024-05-19 12:57:57.431 INFO     PID: 12672 leapp.workflow.Checks: Executing actor check_oracle_enabled_repos 
2024-05-19 12:57:57.497 INFO     PID: 17624 leapp.workflow.Checks.check_oracle_enabled_repos: Setting Oracle enabled repos to None
2024-05-19 12:57:57.509 ERROR    PID: 12672 leapp.workflow.Checks: Actor check_oracle_enabled_repos has crashed: Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/leapp/repository/actor_definition.py", line 74, in _do_run
    actor_instance.run(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/leapp/actors/__init__.py", line 289, in run
    self.process(*args)
  File "/etc/leapp/repos.d/system_upgrade/common/actors/checkoracleenabledrepos/actor.py", line 26, in process
    self.produce(OracleEnabledRepos(enabled_repos=leapp_enabled_repos))
  File "/usr/lib/python3.6/site-packages/leapp/models/__init__.py", line 90, in __init__
    getattr(defined_fields[field], init_method)(kwargs, field, self)
  File "/usr/lib/python3.6/site-packages/leapp/models/fields/__init__.py", line 111, in from_initialization
    self._validate_model_value(value=source_value, name=name)
  File "/usr/lib/python3.6/site-packages/leapp/models/fields/__init__.py", line 180, in _validate_model_value
    super(BuiltinField, self)._validate_model_value(value, name)
  File "/usr/lib/python3.6/site-packages/leapp/models/fields/__init__.py", line 61, in _validate_model_value
    raise ModelViolationError('The value of "{name}" field is None, but this is not allowed'.format(name=name))
leapp.models.fields.ModelViolationError: The value of "enabled_repos" field is None, but this is not allowed

2024-05-19 12:57:57.558 INFO     PID: 12672 leapp: Answerfile will be created at /var/log/leapp/answerfile
2024-05-19 12:57:57.567 WARNING  PID: 12672 leapp.reporting: Stable Key report entry not provided, dynamically generating one - 72a65b379954a49ef1ec64b5937c698795c423d4
2024-05-19 12:57:57.576 WARNING  PID: 12672 leapp.reporting: Stable Key report entry not provided, dynamically generating one - 72a65b379954a49ef1ec64b5937c698795c423d4
```

OS: Oracle 8.9
Kernel: 5.15.0-206.153.7.el8uek.x86_64
Repository configs: 
`/etc/yum.repos.d/oracle-epel-ol8.repo`
```
[ol8_developer_EPEL]
name=Oracle Linux $releasever EPEL Packages for Development ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/developer/EPEL/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=1

exclude=nagios-plugins-2.3.3
[ol8_developer_EPEL_modular]
name=Oracle Linux $releasever EPEL Modular Packages for Development ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/developer/EPEL/modular/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=1
```

`/etc/yum.repos.d/oracle-epel-ol8.repo`
```
[ol8_baseos_latest]
name=Oracle Linux 8 BaseOS Latest ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/baseos/latest/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=1

[ol8_appstream]
name=Oracle Linux 8 Application Stream ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/appstream/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=1

[ol8_codeready_builder]
name=Oracle Linux 8 CodeReady Builder ($basearch) - Unsupported
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/codeready/builder/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=1

[ol8_distro_builder]
name=Oracle Linux 8 Distro Builder ($basearch) - Unsupported
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/distro/builder/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0


[ol8_u0_baseos_base]
name=Oracle Linux 8 BaseOS GA ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/0/baseos/base/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol8_u1_baseos_base]
name=Oracle Linux 8.1 BaseOS ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/1/baseos/base/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol8_u2_baseos_base]
name=Oracle Linux 8.2 BaseOS ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/2/baseos/base/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol8_u3_baseos_base]
name=Oracle Linux 8.3 BaseOS ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/3/baseos/base/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol8_u4_baseos_base]
name=Oracle Linux 8.4 BaseOS ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/4/baseos/base/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol8_u5_baseos_base]
name=Oracle Linux 8.5 BaseOS ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/5/baseos/base/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol8_u6_baseos_base]
name=Oracle Linux 8.6 BaseOS ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/6/baseos/base/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol8_u7_baseos_base]
name=Oracle Linux 8.7 BaseOS ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/7/baseos/base/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol8_u8_baseos_base]
name=Oracle Linux 8.8 BaseOS ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/8/baseos/base/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol8_u9_baseos_base]
name=Oracle Linux 8.9 BaseOS ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/9/baseos/base/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol8_addons]
name=Oracle Linux 8 Addons ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/addons/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol8_u4_security_validation]
name=Oracle Linux $releasever Update 4 ($basearch) Security Validations
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/4/security/validation/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol8_MODRHCK]
name=Latest RHCK with fixes from Oracle for Oracle Linux $releasever ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/MODRHCK/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
priority=20
enabled=0
```

`/etc/yum.repos.d/uek-ol8.repo`

```
[ol8_UEKR7]
name=Latest Unbreakable Enterprise Kernel Release 7 for Oracle Linux $releasever ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/UEKR7/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=1

[ol8_UEKR6]
name=Latest Unbreakable Enterprise Kernel Release 6 for Oracle Linux $releasever ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/UEKR6/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol8_UEKR7_RDMA]
name=Oracle Linux 8 UEK7 RDMA ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/UEKR7/RDMA/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol8_UEKR6_RDMA]
name=Oracle Linux 8 UEK6 RDMA ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/UEKR6/RDMA/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0
```

`/etc/yum.repos.d/virt-ol8.repo`
```
[ol8_kvm_appstream]
name=Oracle Linux $releasever KVM Application Stream ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL8/kvm/appstream/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0
```

Upon running `leapp preupgrade`, a new repository file is created.
`/etc/yum.repos.d/leapp-upgrade-repos-ol9.repo`
```
[ol9_baseos_latest]
name=Oracle Linux 9 BaseOS Latest  ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL9/baseos/latest/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol9_appstream]
name=Oracle Linux 9 Application Stream Packages ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL9/appstream/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol9_addons]
name=Oracle Linux 9 Addons ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL9/addons/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol9_codeready_builder]
name=Oracle Linux 9 CodeReady Builder ($basearch) - (Unsupported)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL9/codeready/builder/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol9_UEKR7]
name=Oracle Linux 9 UEK Release 7 ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL9/UEKR7/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol9_RDMA]
name=Oracle Linux 9 ($basearch) RDMA
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL9/RDMA/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol9_kvm_utils]
name=Oracle Linux 9 KVM Utilities ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL9/kvm/utils/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
priority=10
enabled=0

[ol9_oci_included]
name=Oracle Linux $releasever OCI Included Packages ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL9/oci/included/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol9_ksplice]
name=Ksplice for Oracle Linux $releasever ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL9/ksplice/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0

[ol9_oraclelinuxmanager210_client]
name=Oracle Linux Manager Client 2.10 for Oracle Linux 9 ($basearch)
baseurl=https://yum$ociregion.$ocidomain/repo/OracleLinux/OL9/oraclelinuxmanager210/client/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=0
```

**Comment by aburmash** [comment]
I am making an educated guess that you have executed just
`leapp preupgrade`
If that is so, could you please follow documentation:
https://docs.oracle.com/en/operating-systems/oracle-linux/9/leapp

It specifically suggests to run


```
Use the appropriate command argument for a system or an Oracle Cloud Infrastructure instance.

On a system:
`sudo leapp preupgrade --oraclelinux [--enablerepo repository]`


On an instance in Oracle Cloud Infrastructure:
`sudo leapp preupgrade --oci [--enablerepo repository]`

For detailed information about the arguments, see [Using Command Arguments to Enable Repositories](https://docs.oracle.com/en/operating-systems/oracle-linux/9/leapp/leapp-SupportedRepositoriesinLeappUpgrades.html#repo-auto-enabling).

This process generates a process log, a report, and a file called answerfile.

```
So you need to set either --oraclelinux or --oci depending on where you run the upgrade to enable minimal needed set of repositories.
I also suggest to read other steps that should be done before running the upgrade, as missing them might affect your experience with leapp.

**Comment by ku4eto** [comment]
> I am making an educated guess that you have executed just `leapp preupgrade` If that is so, could you please follow documentation: https://docs.oracle.com/en/operating-systems/oracle-linux/9/leapp
> 
> It specifically suggests to run
> 
> ```
> Use the appropriate command argument for a system or an Oracle Cloud Infrastructure instance.
> 
> On a system:
> `sudo leapp preupgrade --oraclelinux [--enablerepo repository]`
> 
> 
> On an instance in Oracle Cloud Infrastructure:
> `sudo leapp preupgrade --oci [--enablerepo repository]`
> 
> For detailed information about the arguments, see [Using Command Arguments to Enable Repositories](https://docs.oracle.com/en/operating-systems/oracle-linux/9/leapp/leapp-SupportedRepositoriesinLeappUpgrades.html#repo-auto-enabling).
> 
> This process generates a process log, a report, and a file called answerfile.
> ```
> 
> So you need to set either --oraclelinux or --oci depending on where you run the upgrade to enable minimal needed set of repositories. I also suggest to read other steps that should be done before running the upgrade, as missing them might affect your experience with leapp.

A staggering oversight by my side. Passed that part, now blows up with something different.
Before closing the issue - is `leapp upgrade` with `systemd-networkd` and `systemd-resolved` supported (with NetworkManager removed)? Does not seem to be specified in the documentation and asking before opening a new (useless) issue may be a better idea.

**Comment by aburmash** [comment]
It is not specifically stated as unsupported, though i almost definitely can say that the default answer to reporting that issue would be "why did you remove it" ?

is there any chance you can describe your use-case and what issue you are hitting ? If it is not something super hacky, we might be able to patch leapp to work with it just fine.

**Comment by ku4eto** [comment]
Will open a separate issue, to not mix it up with my embarrassing oversight in case it turns out be an actual issue.
